### PR TITLE
Oauth revoke endpoint

### LINKED ISSUE #1247: Revocation with OAuth Workflow doesn't work with Spring Boot Starter
Tested in versions 1.2.5 and 1.3.1.

When trying to logout using OAuth and the latest version of the Java SDK, a POST to `/oauth/revoke` fails.

```
Request URL: http://localhost:8080/oauth/revoke
Request Method: POST
Status Code: 401 
Remote Address:[::1]: 8080
```

Response is:

```json
{
    "status":"error.accessDenied",
    "message":"Full authentication is required to access this resource",
    "fieldErrors":null
}
```

I tried changing `StormpathSecurityConfigurerAdapter.java` to allow access to “/oauth/revoke”. 

```java
if (accessTokenEnabled) {
    http.authorizeRequests().antMatchers(accessTokenUri).permitAll();
    http.authorizeRequests().antMatchers("/oauth/revoke").permitAll();
}
```

However, it seems this endpoint isn’t mapped.

```
Request URL: http://localhost:8080/oauth/revoke
Request Method: POST
Status Code: 404 
Remote Address:[::1]: 8080
```

It is mapped in Client API and express-stormpath.

Spec is in "Token+Authentication+and+Management" in Confluence.

### PR DESCRIPTION
**Pull request by mrioan**
Resolves #1247 

UAT: Can you please check that `/oauth/revoke` works fine in servlet?

Generate Access Token

```
curl -X POST \
-H "Content-Type: application/x-www-form-urlencoded" \
"http://localhost:8080/oauth/token?grant_type=password&username=XXXXXX&password=YYYY"
```

Revoke: 
```
curl -X POST http://localhost:8080/oauth/revoke \
-H "Content-Type: application/x-www-form-urlencoded" \
-d "token=ACCESS_OR_REFRESH_TOKEN_HERE"
```

Check the token is not valid any longer either via Admin Console or with this command (should fail):

```
curl -v -L -X POST \
-H "Accept: application/json" \
-H "Cache-Control: no-cache" \
-H "Authorization: Bearer ACCESS_TOKEN_HERE" \
"http://localhost:8080/logout"
```

### CODE REVIEW

**Comment by mraible** [comment]
@mrioan I tried this branch today and I found that I was unable to POST to `/oauth/token`. The response is:

```json
{"status":"error.accessDenied","message":"Full authentication is required to access this resource","fieldErrors":null}
```

**Comment by mraible** [comment]
@mrioan I tried adding the following `SecurityConfiguration.java` to my generated project and it works now.

```java
package com.example;

import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;

@Configuration
public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
    @Override
    protected void configure(HttpSecurity http) throws Exception {
    }
}
```

👍  LGTM!

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/test/groovy/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfigurationIT.groovy [hunk 1]
```diff
@@ -232,8 +232,9 @@ class StormpathWebSecurityAutoConfigurationIT extends AbstractTestNGSpringContex
         def accessToken = result.getAccessToken()
 
         expect(httpServletRequest.getHeader("Authorization")).andReturn("Bearer " + accessToken.getJwt())
-        expect(httpServletRequest.getServletContext()).andReturn(servletContext).times(2)
-        expect(servletContext.getAttribute("com.stormpath.sdk.client.Client")).andReturn(client).times(2)
+        //expect(httpServletRequest.getServletContext()).andReturn(servletContext).times(2)
+        expect(httpServletRequest.getAttribute(Application.class.getName())).andReturn(application).times(1)
+        //expect(servletContext.getAttribute("com.stormpath.sdk.client.Client")).andReturn(client).times(2)
```
Comment (mraible) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/test/groovy/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfigurationIT.groovy]: Remove commented code.

File: extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 2]
```diff
@@ -387,6 +387,12 @@
       "description": "This is only used for additional origin URIs that don't already match the server URI."
     },
     {
+      "name": "stormpath.web.oauth2.revoke.uri",
+      "type": "java.lang.String",
+      "description": "The context-relative path an HTTP client may POST to revoke access and refresh tokens.",
```
Comment (mraible) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Should this be "may POST to to revoke"?

Comment (mraible) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Maybe it'd be better to say "may POST to in order to..."

File: extensions/spring/stormpath-spring-security-webmvc/src/test/groovy/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcConfigurationIT.groovy [hunk 3]
```diff
@@ -206,8 +211,8 @@ class MinimalStormpathSpringSecurityWebMvcConfigurationIT extends AbstractClient
         def accessToken = result.getAccessToken()
 
         expect(httpServletRequest.getHeader("Authorization")).andReturn("Bearer " + accessToken.getJwt())
-        expect(httpServletRequest.getServletContext()).andReturn(servletContext).times(2)
-        expect(servletContext.getAttribute("com.stormpath.sdk.client.Client")).andReturn(client).times(2)
+        //expect(httpServletRequest.getServletContext()).andReturn(servletContext).times(2)
```
Comment (mraible) [inline extensions/spring/stormpath-spring-security-webmvc/src/test/groovy/com/stormpath/spring/config/MinimalStormpathSpringSecurityWebMvcConfigurationIT.groovy]: Remove commented code.

File: extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/test/groovy/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfigurationIT.groovy (hunk 1 above)
Comment (mrioan) [inline extensions/spring/boot/stormpath-spring-security-webmvc-spring-boot-starter/src/test/groovy/com/stormpath/spring/boot/autoconfigure/StormpathWebSecurityAutoConfigurationIT.groovy]: Done

File: extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 2 above)
Comment (mrioan) [inline extensions/spring/boot/stormpath-webmvc-spring-boot-starter/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: Done

**Comment by mrioan** [comment]
@mraible I applied all your requested changed. Can this PR be considered approved?

**Comment by mraible** [comment]
Yep!

**Pull request merged**